scylla/src/transport/session.rs

```diff
@@ -409,9 +410,25 @@ impl Session {
         &self,
         prepared: &PreparedStatement,
         values: impl ValueList,
+    ) -> Result<QueryResult, QueryError> {
+        self.execute_paged(prepared, values, None).await
+    }
+
+    /// Executes a previously prepared statement with previously received paging state
+    /// # Arguments
+    ///
+    /// * `prepared` - a statement prepared with [prepare](crate::transport::session::Session::prepare)
+    /// * `values` - values bound to the query
+    /// * `paging_state` - paging state from the previous query or None
+    pub async fn execute_paged(
```

Comment:
Have you considered adding a `paging_state` field to `Query` and `PreparedStatement`? My main worry here is that if we decide to add another optional parameter like that, you will need to have a separate method for each combination of "with parameter"/"without parameter", or force the user to always provided `None` or something like that.

On the other hand, the problem with `paging_state` field is that somebody might forget that they set it...

Reply:
I did, but I decided it doesn't belong to any particular query or statement. It's just a way of resuming a previous run of a particular query, so this value is expected to change on each call to execute()/query()

Reply:
I see. In that case, I think the current interface is OK.